nutridine/src/hooks/useMealsByMacros.js

```diff
@@ -4,88 +4,95 @@ import brandIds from "../constants/brandIds";
 
 // DOCS: https://docx.syndigo.com/developers/docs/instant-endpoint
 /** All parameters of type number (except "query", which is a string). Assumed units for all macros are grams */
-const useMealsByMacros = ({
-  query,
-  minCalories,
-  maxCalories,
-  minProtein,
-  maxProtein,
-  minCarbs,
-  maxCarbs,
-  minFat,
-  maxFat,
-}) => {
+const useMealsByMacros = (
+  {
+    query,
+    minCalories,
+    maxCalories,
+    minProtein,
+    maxProtein,
+    minCarbs,
+    maxCarbs,
+    minFat,
+    maxFat,
+  },
+  shouldFetch
+) => {
   const [meals, setMeals] = useState([]);
   const [loading, setLoading] = useState(true);
   const [error, setError] = useState(null);
 
-  useEffect(() => {
-    const apiKey = process.env.REACT_APP_NUTRITIONIX_API_KEY;
-    const appId = process.env.REACT_APP_NUTRITIONIX_APP_ID;
+  useEffect(
+    () => {
+      const apiKey = process.env.REACT_APP_NUTRITIONIX_API_KEY;
+      const appId = process.env.REACT_APP_NUTRITIONIX_APP_ID;
 
-    const fetchData = async () => {
-      try {
-        // Requires a POST request to filter by calories.
-        // Cannot inject variables in json field (for nutritionIX food ids)
-        // If you omit the query, nutritionIX rejects the request.
-        let response = await axios.post(
-          "https://trackapi.nutritionix.com/v2/search/instant",
-          {
-            query: Object.keys(brandIds).join(" ") + " " + query,
-            brand_ids: Object.values(brandIds),
```

Comment:
this looks like a regression. Can you make sure we get the old version that filters based on Victoria please?